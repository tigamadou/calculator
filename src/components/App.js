import React, { Component } from 'react';

import Display from './Display';
import ButtonPanel from './ButtonPanel';
// eslint-disable-next-line no-unused-vars
import Calculate from '../logic/calculate';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const newState = Calculate(this.state, buttonName);
    this.setState(() => ({ ...newState }));
  }

  render() {
    const { total } = this.state;
    return (
      <>
        <Display result={total} />
        <ButtonPanel handleClick={this.handleClick} />
      </>
    );
  }
}

export default App;
