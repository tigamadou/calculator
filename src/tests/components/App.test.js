import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../../components/App';

describe('App', () => {
  test('renders App component ', () => {
    render(<App />);
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('Letâ€™s do some math!')).toBeInTheDocument;
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
