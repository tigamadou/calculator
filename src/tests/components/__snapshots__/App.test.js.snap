// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match with snapshot 1`] = `
<div
  className="calculatorPage"
>
  <div>
    <h2>
      Let’s do some math!
    </h2>
  </div>
  <div
    className="calculatorApp"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="panel"
    >
      <div
        className="row"
      >
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          AC
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          +/-
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          %
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          ÷
        </div>
      </div>
      <div
        className="row"
      >
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          7
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          8
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          9
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          x
        </div>
      </div>
      <div
        className="row"
      >
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          4
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          5
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          6
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          -
        </div>
      </div>
      <div
        className="row"
      >
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          1
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          2
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          3
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          +
        </div>
      </div>
      <div
        className="row"
      >
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          0
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          .
        </div>
        <div
          aria-hidden="true"
          className="button"
          onClick={[Function]}
          role="button"
        >
          =
        </div>
      </div>
    </div>
  </div>
</div>
`;
